### DISCUSSION TITLE
Possible conflict of Inovarti Platform with OpenMage License?

### DISCUSSION THREAD

**Discussion by empiricompany** [OpenMage/magento-lts#3581] [General]
What do you think about this? 
https://github.com/OpenMage/magento-lts/discussions/3458#discussioncomment-7199620

https://www.inovarti.com.br/inovarti-platform-e-atualizacoes-para-2021/
Is this a potential violation of the OpenMage license? It seems to me that it's a fork of the core not publicly released

@fballiano @Flyingmana @sreichel @kiatng @colinmollenhour @ADDISON74 @drobinson @LeeSaferite @elidrissidev @tmotyl

**Comment by elidrissidev** [comment]
Legal matters are not my expertise, but if all forks were to be required to be open-source, doesn't that mean that technically anyone with a store that uses OM is using a fork and therefore is required to make it open-source?

**Comment by addison74** [comment]
It seems the owner is reading the discussion in this repository. In my case accessing that link it says "Access denied".

**Reply by elidrissidev** [reply]
It works for me, it's an old post from 2020.

**Reply by empiricompany** [reply]
yes it's work also for me from Italy / Germany IPs

**Reply by fballiano** [reply]
me too from spain

**Reply by addison74** [reply]
I will change the VPN or access it through an online proxy like proxysite.com.

**Comment by empiricompany** [comment]
I'm also not an expert on licenses, but if you profit from a fork that basically modifies the core, is this legit?
I ask because I thought that the core would remain under the license, so all the stores are in compliance if they don't modify it or profit from a fork based on it in any way.

**Reply by addison74** [reply]
The legal issue is a bit more complicated. First of all there is Magento which is the property of a company. OpenMage is based on Magento's open source code and came with an obviously smaller weight of changes. OpenMage does not belong to a company but is a project maintained by a community that again has no legal status. If that company makes money with OpenMage how do you prove ownership and infringement? Is it just that there is a copyright line with OpenMage in certain files? OpenMage is not even a registered trademark as far as I know ...

**Reply by empiricompany** [reply]
Honestly, I don't know. I was asking to learn more about it

**Comment by fballiano** [comment]
AFAIK OSL3 requires you to release a forked project in OSL3 too, but that doesn't mean you have to make it freely available to everybody. AFAIK that's why it has been chosen over GPL/AGPL.

I asked chatgpt for confirmation:

> The Open Software License 3.0 (OSL-3.0) requires that if you create a fork or derivative work based on OSL-3.0-licensed software, you must release that derivative work as open source under the same OSL-3.0 license or a compatible open-source license. However, it doesn't specifically mandate that the fork or derivative work has to be publicly available for free.
> 
> The OSL-3.0 focuses on ensuring that the software remains open source and that others have the same rights to use, modify, and distribute it as you did. It doesn't prescribe whether you can charge for access to the software or related services. You can still charge for your forked or modified version of the software, but you must provide access to the source code under the terms of the OSL-3.0.
> 
> In other words, you can commercialize OSL-3.0-licensed software, but you must grant others the same rights and access to the source code that you received under the OSL-3.0 license. This allows for both open-source collaboration and potential commercialization of the software.

**Reply by empiricompany** [reply]
So if I purchase their fork based on OM, in this case Inovarti Platform, do I have the right to access all the source code and can I also create/copy the code and release it publicly or profit from it myself?

**Reply by fballiano** [reply]
yes, and that's why magento enterprise, back in the day, had a completely different copyright code in every single file, also that ones what were in common with community edition

**Reply by empiricompany** [reply]
and that's why we can't porting the features developed in Magento Enterprise, right?

**Reply by fballiano** [reply]
exactly, that code is "copyright by magento (or adobe, I don't remember)" and it's not open source

**Reply by fballiano** [reply]
I mean, it is open source because they were giving you the code, but with a copyright license, so nobody can copy anything from there or redistribute anything.

**Reply by Flyingmana** [reply]
chatgpt is a terrible choice for legal advise, there is no guarantee any part of what it says is correct.

If you want to have a good summary, take this website as a first check https://www.tldrlegal.com/license/open-software-licence-3-0

Besides this, yes OSl-3 does not require to make the sourcecode public of software which is used to provide a website.
There was a decade ago a big discussion, about the whole topic, there are only very few licenses which require to make the source code for websites public. Following https://itmedialaw.com/en/open-source-licenses-gpl-agpl-mit-and-apache/ this apparently is only really true for the AGPL license.

**Reply by fballiano** [reply]
I had a pretty famous post on my blog about this specific topic, also Roy Rubin commented heheheh

At that time I was promoting AGPL but it's was to utopistically FLOSS

**Reply by Flyingmana** [reply]
Fun thing, (after getting access to the blog post over the wayback machine), mentioning our name in the blogpost could be seen as a license violation (if we would have registered it as a trademark), but thats very far fetched

**Reply by fballiano** [reply]
I can't find it and I don't remember the details, it was too many years ago.
But also I didn't understand what you said about the trademark.

**Reply by Flyingmana** [reply]
The OSL-3 license does not allow to use `contributors' names, trademarks or logos` to "advertise" for your own software.

**Comment by empiricompany** [comment]
thanks all, i'm thinking that indeed that's why projects like MageOne are perfectly legit

**Reply by Flyingmana** [reply]
MageOne did approach it differently, they only provided `.patch` files, not actually changed source code.

**Reply by empiricompany** [reply]
so basically you that apply that patch to the core create a "forked patched version"

**Comment by Flyingmana** [comment]
What do I think about it.
I think its great its of use for them.
Its great that they did see OpenMage at the time as the best base to start their own fork.

But maintaining a fork long enough without contributing back becomes expensive,
as the post is from 2021, before OpenMage started to move faster, they likely did not take over much of the changes we have done in the last 2 years, or will at one point realize its easier to move the biggest changes into own modules, or back to OpenMage. So I dont think they are getting an unjust advantage, if anyone wonders about it.

The license part is already answered enough I think 😅

**Comment by LuizSantos22** [comment]
@empiricompany I have investigated and I found out that Inovarti Theme (now Inovarti V3) is actually based on Ultimo (probably with Bootstrap 4), most of the jobs they do are based on Ultimo Store 5, highly modified, but still a lot of Ultimo's callouts, tips and features are still there, like the expanding header menu on top which expands when you scroll down. 

I was thinking about something recreating this theme like "Ultimo Refactored" and implement changes or a skeleton to make it look like Inovarti's theme, which is quite nice and Ultimo is a very easy to customize theme. 

I guess they've used Ultimo as a base and them, started upgrading it. There's another site they've worked called "kaisan" and it is Ultimo, but with some modifications...

I guess OpenMage needs a "pretty" frontend theme so it will encorage users to use it and keep their theme up to date... Maybe it is just a dream, but I really like this theme because once you get use to it, you have a lot of features to make customizations.
